plugins/nominal-connection-checker/src/type_hierarchy.js

```diff
@@ -309,6 +448,186 @@ export class TypeHierarchy {
           struct.name, struct.params.length, def.params().length);
     }
   }
+
+  /**
+   * Returns the names of the nearest common types of the types associated with
+   * the given type names, as defined by the passed type map. Does not handle
+   * parameterized types! That is handled by getNearestCommonParents and
+   * getNearestCommonAncestors.
+   * @param {!Array<string>} typeNames The types to find the nearest common
+   *     type of.
+   * @param {!Map<string, !Map<string, Array<string>>>} nearestCommon A map
+   *     defining the nearest common types of each pair of types.
+   * @return {!Array<string>} The names of the nearest common types of the types
+   *     associated with the given type names.
+   * @private
+   */
+  getNearestCommon_(typeNames, nearestCommon) {
+    return typeNames.reduce((accumulator, currType) => {
+      const commonMap = nearestCommon.get(currType);
+      // Note: neither flatMap() nor flat() work on Node 10. See #431.
+      return accumulator
+          .map((type) => {
+            return [...commonMap.get(type)]; // Create copy to avoid corruption.
+          })
+          .reduce((flat, toFlatten) => {
+            return [...flat, ...toFlatten];
+          }, [])
```

Comment:
Could you further explain what's happening here, maybe add a comment in the reduce function.

Reply:
I had to look up `flat()` and `flatMap()` since it's not a common function I use. 
I think the comment is helpful, but a comment by the reduce function would make comprehension easier, though not strictly necessary.
Something like  "Flatten the array of arrays" on L474 (or add Arr or Array to the variables of the reduce function as a reminder of the types)